### PR TITLE
[c10d] Remove health check

### PR DESCRIPTION
**Pull request by kwen2501**
https://github.com/pytorch/pytorch/pull/114916 and https://github.com/pytorch/pytorch/pull/116222 added support for eager NCCL comm init (performed as soon as `init_process_group` is called).

If any user cares about the time difference and want to see NCCL init errors early, they can use eager init now.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

**Comment by wconstab** [comment]
is your point that health check was _only_ useful for flushing out initialization errors, and if users want health check they should use eager init instead?

maybe be clearer (in the api docs) about how to achieve this but i think it makes sense.

**Comment by kwen2501** [comment]
> is your point that health check was _only_ useful for flushing out initialization errors, and if users want health check they should use eager init instead?

Yes

> maybe be clearer (in the api docs) about how to achieve this but i think it makes sense.

Good idea, updated!

**Comment by wconstab** [comment]
if we had any docs (?) for health check, i would suggest updating those to point to this too.  But maybe we didn't.

**Review by wconstab — APPROVED** [review]

**Comment by kwen2501** [comment]
I did searched for "health" and did not find any left.

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 1]
```diff
@@ -956,65 +947,6 @@ c10::intrusive_ptr<intra_node_comm::IntraNodeComm> ProcessGroupNCCL::
       store_, std::to_string(uid_), rank_, size_);
 }
 
-void ProcessGroupNCCL::runHealthCheck() {
```
Comment (fduwjj) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:959]: oh this is not used. I always have questions about how it is used.

**Pull request closed without merging**